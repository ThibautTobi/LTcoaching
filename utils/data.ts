export type dataType = {
  id: string;
  title: string;
  image: string;
  imageAlt: string;
  Description: string;
  price: string;
  showLink?: boolean;
  showImage?: boolean;
  categories: ('sport' | 'bien-être' | 'nutrition')[];
};

export const DataService: dataType[] = [
  // {
  //   id: 'coaching-sportif-1',
  //   title: 'Coaching Sportif Personnalisé',
  //   image: 'src',
  //   imageAlt: 'src3',
  //   Description: 'Programme d’entraînement adapté à vos objectifs.',
  //   price: '50€ / séance',
  //   categories: ['sport'],
  // },
  {
    id: 'coaching-nutrition-1',
    title: 'Coaching Nutritionnel',
    image: '/nutrition.jpg',
    imageAlt: 'src3',
    Description: 'Plan alimentaire personnalisé et suivi hebdomadaire.',
    price: '200€ / Plan Nutritionnel',
    categories: ['nutrition'],
  },
  // {
  //   id: 'coaching-en-ligne-1',
  //   title: 'Coaching en ligne',
  //   image: 'src',
  //   imageAlt: 'src3',
  //   Description: 'Suivi à distance via une application dédiée.',
  //   price: '40€ / séance',
  //   categories: ['sport'],
  // },
  //   {
  //   id: 'coaching-en-ligne-2',
  //   title: 'Coaching en ligne',
  //   image: 'src',
  //   imageAlt: 'src3',
  //   Description: 'Suivi à distance via une application dédiée.',
  //   price: '40€ / séance',
  //   categories: ['bien-être'],
  // },
];
