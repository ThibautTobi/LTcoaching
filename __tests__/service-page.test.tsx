// import React from 'react';
// import { render, screen, fireEvent } from '@testing-library/react';
// import ServicesPage from '@/app/Services/page';

// // // On mock DataService
// // jest.mock('@/utils/data', () => ({
// //   DataService: [
// //     { id: '1', title: 'Coaching Sportif', categories: ['sport'] },
// //     { id: '2', title: 'Massage Bien-Ãªtre', categories: ['bien-Ãªtre'] },
// //     { id: '3', title: 'Conseil Nutrition', categories: ['nutrition'] },
// //     { id: '4', title: 'Pack Complet', categories: ['sport', 'nutrition'] },
// //   ],
// // }));

// import { dataType } from '@/utils/data';

// jest.mock('@/utils/data', () => ({
//   DataService: [
//     {
//       id: '1',
//       title: 'Coaching Sportif',
//       categories: ['sport'],
//       image: '/sport.jpg',
//       imageAlt: 'Coaching Sportif',
//       Description: 'Un coaching personnalisÃ©',
//       liste: ['Objectif', 'RÃ©sultat'],
//       price: '50',
//     },
//     {
//       id: '2',
//       title: 'Programme Nutrition',
//       categories: ['nutrition'],
//       image: '/nutrition.jpg',
//       imageAlt: 'Programme Nutrition',
//       Description: 'Plan alimentaire sur mesure',
//       liste: ['Suivi', 'Plan'],
//       price: '40',
//     },
//   ] satisfies dataType[],
// }));

// // On mock CardService pour simplifier lâ€™affichage
// jest.mock('@/app/src/components/card', () => ({
//   CardService: (props: any) => (
//     <div data-testid="card">{props.DataService.title}</div>
//   ),
// }));

// // On mock Button pour Ã©viter les styles complexes
// jest.mock('@/src/components/ui/button', () => ({
//   Button: (props: any) => <button {...props}>{props.children}</button>,
// }));

// describe('ServicesPage Component', () => {
//   it('affiche le titre principal', () => {
//     render(<ServicesPage />);
//     expect(
//       screen.getByRole('heading', {
//         name: /Nos Services de Coaching & Nutrition/i,
//       })
//     ).toBeInTheDocument();
//   });

//   it('affiche tous les boutons de filtre', () => {
//     render(<ServicesPage />);
//     const filters = ['tous', 'sport', 'bien-Ãªtre', 'nutrition'];
//     filters.forEach((filter) => {
//       expect(
//         screen.getByRole('button', { name: new RegExp(filter, 'i') })
//       ).toBeInTheDocument();
//     });
//   });

//   it('affiche tous les services au chargement', () => {
//     render(<ServicesPage />);
//     expect(screen.getAllByTestId('card')).toHaveLength(4);
//   });

//   it('filtre les services par catÃ©gorie', () => {
//     render(<ServicesPage />);

//     // On clique sur "sport"
//     fireEvent.click(screen.getByRole('button', { name: /sport/i }));

//     // On ne doit avoir que ceux avec catÃ©gorie "sport"
//     const visibleCards = screen
//       .getAllByTestId('card')
//       .map((el) => el.textContent);
//     expect(visibleCards).toEqual(
//       expect.arrayContaining(['Coaching Sportif', 'Pack Complet'])
//     );
//     expect(visibleCards).toHaveLength(2);

//     // On clique sur "bien-Ãªtre"
//     fireEvent.click(screen.getByRole('button', { name: /bien-Ãªtre/i }));
//     const bienEtreCards = screen
//       .getAllByTestId('card')
//       .map((el) => el.textContent);
//     expect(bienEtreCards).toEqual(['Massage Bien-Ãªtre']);

//     // On clique sur "tous"
//     fireEvent.click(screen.getByRole('button', { name: /tous/i }));
//     expect(screen.getAllByTestId('card')).toHaveLength(4);
//   });
// });
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import ServicesPage from '@/app/Services/page';
import { dataType } from '@/utils/data';

// --- ðŸ”¹ MOCKS ---
// Mock de DataService
jest.mock('@/utils/data', () => ({
  DataService: [
    {
      id: '1',
      title: 'Coaching Sportif',
      categories: ['sport'],
      image: '/sport.jpg',
      imageAlt: 'Coaching Sportif',
      Description: 'Un coaching personnalisÃ©',
      liste: ['Objectif', 'RÃ©sultat'],
      price: '50',
    },
    {
      id: '2',
      title: 'Massage Bien-Ãªtre',
      categories: ['bien-Ãªtre'],
      image: '/massage.jpg',
      imageAlt: 'Massage Bien-Ãªtre',
      Description: 'Un moment de dÃ©tente',
      liste: ['Relaxation', 'Bien-Ãªtre'],
      price: '60',
    },
    {
      id: '3',
      title: 'Conseil Nutrition',
      categories: ['nutrition'],
      image: '/nutrition.jpg',
      imageAlt: 'Conseil Nutrition',
      Description: 'Des conseils alimentaires adaptÃ©s',
      liste: ['Ã‰quilibre', 'SantÃ©'],
      price: '40',
    },
    {
      id: '4',
      title: 'Pack Complet',
      categories: ['sport', 'nutrition'],
      image: '/pack.jpg',
      imageAlt: 'Pack Complet',
      Description: 'Sport + Nutrition pour des rÃ©sultats rapides',
      liste: ['Programme sport', 'Suivi nutritionnel'],
      price: '90',
    },
  ] satisfies dataType[],
}));

// Mock de CardService
jest.mock('@/app/src/components/card', () => ({
  CardService: ({ DataService }: { DataService: dataType }) => (
    <div data-testid="card">{DataService.title}</div>
  ),
}));

// Mock du bouton UI
jest.mock('@/src/components/ui/button', () => ({
  Button: ({
    children,
    ...rest
  }: React.ButtonHTMLAttributes<HTMLButtonElement> & {
    children: React.ReactNode;
  }) => <button {...rest}>{children}</button>,
}));

// --- ðŸ”¹ TESTS ---
describe('ServicesPage Component', () => {
  it('affiche le titre principal', () => {
    render(<ServicesPage />);
    expect(
      screen.getByRole('heading', {
        name: /Nos Services de Coaching Sportif, Nutrition & Bien-Ãªtre/i,
      })
    ).toBeInTheDocument();
  });

  it('affiche tous les boutons de filtre', () => {
    render(<ServicesPage />);
    const filters = ['tous', 'sport', 'bien-Ãªtre', 'nutrition'];
    filters.forEach((filter) => {
      expect(
        screen.getByRole('button', { name: new RegExp(filter, 'i') })
      ).toBeInTheDocument();
    });
  });

  it('affiche tous les services au chargement', () => {
    render(<ServicesPage />);
    expect(screen.getAllByTestId('card')).toHaveLength(4);
  });

  it('filtre les services par catÃ©gorie', () => {
    render(<ServicesPage />);

    // --- Filtre sport ---
    fireEvent.click(screen.getByRole('button', { name: /sport/i }));
    let visibleCards = screen
      .getAllByTestId('card')
      .map((el) => el.textContent);
    expect(visibleCards).toEqual(
      expect.arrayContaining(['Coaching Sportif', 'Pack Complet'])
    );
    expect(visibleCards).toHaveLength(2);

    // --- Filtre bien-Ãªtre ---
    fireEvent.click(screen.getByRole('button', { name: /bien-Ãªtre/i }));
    visibleCards = screen.getAllByTestId('card').map((el) => el.textContent);
    expect(visibleCards).toEqual(['Massage Bien-Ãªtre']);

    // --- Filtre nutrition ---
    fireEvent.click(screen.getByRole('button', { name: /nutrition/i }));
    visibleCards = screen.getAllByTestId('card').map((el) => el.textContent);
    expect(visibleCards).toEqual(
      expect.arrayContaining(['Conseil Nutrition', 'Pack Complet'])
    );

    // --- Filtre tous ---
    fireEvent.click(screen.getByRole('button', { name: /tous/i }));
    expect(screen.getAllByTestId('card')).toHaveLength(4);
  });

  it('le bouton sÃ©lectionnÃ© doit avoir une classe spÃ©cifique', () => {
    render(<ServicesPage />);
    const sportButton = screen.getByRole('button', { name: /sport/i });
    fireEvent.click(sportButton);
    expect(sportButton).toHaveClass('font-semibold'); // adapte selon ta vraie classe
  });

  it('correspond au snapshot', () => {
    const { asFragment } = render(<ServicesPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
