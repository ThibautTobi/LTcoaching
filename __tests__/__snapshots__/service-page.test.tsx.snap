// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ServicesPage Component correspond au snapshot 1`] = `
<DocumentFragment>
  <section
    class="max-w-5xl mx-auto py-12 px-4"
  >
    <h1
      class="text-[24px] font-bold text-center text-[#C6A35E] mb-8"
    >
      Nos Services de Coaching Sportif, Nutrition & Bien-être
    </h1>
    <p
      class="text-center max-w-2xl mx-auto mb-10 text-gray-300"
    >
      LT Coaching vous accompagne dans votre transformation physique et mentale.
      <br />
      <br />
      Découvrez nos services sur mesure adaptés à vos objectifs :
      <br />
      <br />
      perte de poids, prise de masse, amélioration des performances, bien-être et équilibre alimentaire.
    </p>
    <div
      class="flex justify-center gap-4 mb-8 flex-wrap"
    >
      <button
        aria-pressed="true"
        class="flex items-center gap-2 px-4 py-2 rounded-full border transition duration-200 ease-in-out
          bg-card text-primary font-semibold border-2 border-primary
        "
      >
        <span />
        <span
          class="capitalize"
        >
          tous
        </span>
      </button>
      <button
        aria-pressed="false"
        class="flex items-center gap-2 px-4 py-2 rounded-full border transition duration-200 ease-in-out
          bg-card border-white text-white hover:bg-[#C6A35E]/20
        "
      >
        <span>
          🏋️
        </span>
        <span
          class="capitalize"
        >
          sport
        </span>
      </button>
      <button
        aria-pressed="false"
        class="flex items-center gap-2 px-4 py-2 rounded-full border transition duration-200 ease-in-out
          bg-card border-white text-white hover:bg-[#C6A35E]/20
        "
      >
        <span>
          🧘
        </span>
        <span
          class="capitalize"
        >
          bien-être
        </span>
      </button>
      <button
        aria-pressed="false"
        class="flex items-center gap-2 px-4 py-2 rounded-full border transition duration-200 ease-in-out
          bg-card border-white text-white hover:bg-[#C6A35E]/20
        "
      >
        <span>
          🍏
        </span>
        <span
          class="capitalize"
        >
          nutrition
        </span>
      </button>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
    >
      <div
        id="1"
      >
        <div
          data-testid="card"
        >
          Coaching Sportif
        </div>
      </div>
      <div
        id="2"
      >
        <div
          data-testid="card"
        >
          Massage Bien-être
        </div>
      </div>
      <div
        id="3"
      >
        <div
          data-testid="card"
        >
          Conseil Nutrition
        </div>
      </div>
      <div
        id="4"
      >
        <div
          data-testid="card"
        >
          Pack Complet
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
